packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs

```diff
@@ -366,4 +368,116 @@ mod tests {
             &[&"Friend Of"],
         );
     }
+
+    #[test]
+    fn entity_simple_query() {
+        let mut compiler = SelectCompiler::<Entity>::with_default_fields();
+
+        let filter = Filter::Equal(
+            Some(FilterExpression::Path(EntityQueryPath::Id)),
+            Some(FilterExpression::Parameter(Parameter::Text(Cow::Borrowed(
+                "12345678-ABCD-4321-5678-ABCD5555DCBA",
+            )))),
+        );
+        compiler.add_filter(&filter);
+
+        test_compilation(
+            &compiler,
+            r#"
+            SELECT
+                "entities"."properties",
+                "entities"."entity_id",
+                "entities"."version",
+                "entity_types_0_0"."schema"->>'$id',
+                "entities"."owned_by_id"
+            FROM "entities"
+            JOIN "entity_types" AS "entity_types_0_0"
+              ON "entity_types_0_0"."version_id" = "entities"."entity_type_version_id"
+            WHERE "entities"."entity_id" = $1
+            "#,
+            &[&"12345678-ABCD-4321-5678-ABCD5555DCBA"],
+        );
+    }
+
+    #[test]
+    fn entity_latest_version_query() {
+        let mut compiler = SelectCompiler::<Entity>::with_default_fields();
+
+        let filter = Filter::Equal(
+            Some(FilterExpression::Path(EntityQueryPath::Version)),
+            Some(FilterExpression::Parameter(Parameter::Text(Cow::Borrowed(
+                "latest",
+            )))),
+        );
+        compiler.add_filter(&filter);
+
+        test_compilation(
+            &compiler,
+            r#"
+            WITH "entities" AS (SELECT *, MAX("entities"."version") OVER (PARTITION BY "entities"."entity_id") AS "latest_version" FROM "entities")
+            SELECT DISTINCT
```

Comment:
Question about distinct:
Do we have to be explicit about the rows we want to be distinct on? 
From the PG docs: `If SELECT DISTINCT is specified, all duplicate rows are removed from the result set`, and given that the whole row won't be the same across entries, won't that lead to problems? (meaning we may have to be explicit about `DISTINCT ON ("entities"."version") "entities"."version"`)

Although this will not be an issue because we join on only the latest entity id. Would be problematic for other queries where we want to be distinct for only parts of the returned rows :)

Reply:
I implemented this in #1272

Reply:
Thank you! This could have been a source of bugs, great to get it out of the way :)